### PR TITLE
Update doc about responding to an interactive dialog submission

### PR DESCRIPTION
**Pull request by jasonblais**
Address https://github.com/mattermost/mattermost-developer-documentation/issues/250

### CODE REVIEW

File: source/developer/interactive-dialogs.rst [hunk 1]
```diff
@@ -242,7 +242,18 @@ Moreover, Mattermost also allows the integration itself to perform input validat
 
   {"errors": {"num_between_0_and_10": "Enter a number between 0 and 10."}}
 
-Finally, once the request is submitted, we recommend the integration to respond with a system message or an ephemeral message confirming the submission.
+Finally, once the request is submitted, we recommend the integration to respond with a system message or an ephemeral message confirming the submission. This should be a separate request back to Mattermost once the service has received (and responded to) a submission request from a dialog:
+
+.. code-block:: none
```
Comment (hanzei) [inline source/developer/interactive-dialogs.rst]: This code example was about sending dialog submission reply and this is unrelated to the above topic. Would you please remove it?

File: source/developer/interactive-dialogs.rst [hunk 2]
```diff
@@ -242,7 +242,18 @@ Moreover, Mattermost also allows the integration itself to perform input validat
 
   {"errors": {"num_between_0_and_10": "Enter a number between 0 and 10."}}
 
-Finally, once the request is submitted, we recommend the integration to respond with a system message or an ephemeral message confirming the submission.
+Finally, once the request is submitted, we recommend the integration to respond with a system message or an ephemeral message confirming the submission. This should be a separate request back to Mattermost once the service has received (and responded to) a submission request from a dialog:
```
[the remark this answers is not in the document]
Comment (jasonblais) [reply source/developer/interactive-dialogs.rst]: Can you help with that? I don't know how to do that.

File: source/developer/interactive-dialogs.rst (hunk 1 above)
Comment (jasonblais) [reply source/developer/interactive-dialogs.rst]: Removed

Comment (jasonblais) [reply source/developer/interactive-dialogs.rst]: The change did seem to receive positive feedback from original reporter, so maybe we could keep it but describe it differently?

```
The amendment to the documentation has made it clearer about what the integration needs to do. If possible, a JSON example for the HTTP API (and endpoint) or a reference to the /posts/ephemeral documentation would be perfect.
```

File: source/developer/interactive-dialogs.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (hanzei) [reply source/developer/interactive-dialogs.rst]: Something like:

```suggestion
Finally, once the request is submitted, we recommend the integration to respond with a system message or an ephemeral message confirming the submission. This should be a separate request back to Mattermost once the service has received (and responded to) a submission request from a dialog. This can be done either via [the REST API](https://api.mattermost.com/#tag/posts%2Fpaths%2F~1posts~1ephemeral%2Fpost) or via the [Plugin API](https://developers.mattermost.com/extend/plugins/server/reference/#API.SendEphemeralPost) if you are developing a plugin.
```

File: source/developer/interactive-dialogs.rst (hunk 1 above)
Comment (hanzei) [reply source/developer/interactive-dialogs.rst]: Let's just link to the API endpoint.

**Comment by jasonblais** [comment]
@hanzei Updated per suggestions.
